components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/UserAuthenticationAPI.java

```diff
@@ -63,14 +68,19 @@ public Response getTokens(@PathParam("authCode") String authCode) throws APIInvo
                     .setCode(authCode).buildBodyMessage();
 
             OAuthClient oAuthClient = new OAuthClient(new URLConnectionClient());
-            OAuthAccessTokenResponse oAuthResponse = oAuthClient.accessToken(request);
+            OAuthAccessTokenResponse oAuthResponse = oAuthClient.accessToken(oAuthClientRequest);
             JSONObject jsonObj = new JSONObject(oAuthResponse.getBody());
             Map<Object, Object> responseMap = new HashMap<>();
 
-            responseMap.put(Constants.ACCESS_TOKEN, oAuthResponse.getAccessToken());
+            String[] tokenParts = splitToken(oAuthResponse.getAccessToken());
+            responseMap.put(Constants.ACCESS_TOKEN, tokenParts[0]);
             responseMap.put(Constants.ID_TOKEN, jsonObj.get(Constants.ID_TOKEN));
```

Comment:
Don't we need to split the ID Token as well ? Please confirm with @hasinthaindrajee

Reply:
As we do need the ID token in the front end for logging out redirection, we do not send it as a httponly cookie.

Reply:
Then shall we reomve the splitToken method? We don't need a separate method for this I think.